packages/strapi-plugin-versioning/src/extensions/content-manager/controllers/Versions.ts

```diff
@@ -15,20 +15,13 @@ type Version = {
 module.exports = {
   async listEntityVersions(ctx: Context): Promise<Version[]> {
     const { model, id } = ctx.params;
-    const versionsForAllContentTypes: Content[] = await global.strapi.plugins[
-      "versioning"
-    ].services.versioning.getVersionsForAllConentTypes();
-
-    const versionsForCurrentContentType = versionsForAllContentTypes.filter(
-      (version) => version.content_type == model,
-    );
-
-    const versionsForCurrentId = versionsForCurrentContentType.filter(
-      (version) => version.entity_id === parseInt(id),
+    const service = global.strapi.plugins['versioning'].services.versioning;
+    const versionsForCurrentContentType: Content[] = await service.getVersionsForEntity(
+      model,
+      id,
     );
-
-    return versionsForCurrentId.map((el) => ({
-      content: el.entity,
+    return versionsForCurrentContentType.map((el) => ({
+      content: JSON.parse(el.entity),
```

Comment:
Looks like `el.entity` is already an object, I reverted to:
```
content: el.entity,
```

Reply:
Strange, because we write data to database as json and here we always get string

Reply:
Hey @Kronos66 I tried https://github.com/VirtusLab-Open-Source/strapi-molecules/pull/59/commits/0d5e94c7c127a380ab643ed5854724a191309c03 and I can confirm it fails with `JSON.parse(el.entity)`.

Reply:
I have one question about this problem, what database do you use?

Reply:
postgresql

Reply:
If i remember correctly this happens also with built-in sqlite @Kronos66

Reply:
Hi, @vuolter , sorry for delay 
I added the `saveParse` method to handle objects, now everything should works correctly